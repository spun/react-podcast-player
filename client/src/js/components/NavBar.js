import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

// Componente que muestra una barra de navegación superiór
const NavBar = ({ userEmail }) => {
	return (
		<nav className="navbar navbar-default">
			<div className="container-fluid">
				<div className="navbar-header">
					<button type="button" className="navbar-toggle">
						<span className="sr-only">Toggle navigation</span>
						<span className="icon-bar bar1" />
						<span className="icon-bar bar2" />
						<span className="icon-bar bar3" />
					</button>
					<a className="navbar-brand" href="#">
						Dashboard
					</a>
				</div>
				<div className="collapse navbar-collapse">
					<ul className="nav navbar-nav navbar-right">
						{/*Botón de añadir suscripción*/}
						<li>
							<Link to="/add">
								<i className="ti-plus" /> <p>Add suscription</p>
							</Link>
						</li>

						{/*Menú desplegable de ajustes*/}
						<li className="dropdown">
							<a
								href="#"
								className="dropdown-toggle"
								data-toggle="dropdown">
								<i className="ti-settings" /> <p>Settings</p>
								<b className="caret" />
							</a>
							<ul className="dropdown-menu">
								<li>
									<Link to="/account">My account</Link>
								</li>
								<li>
									<Link to="/about">About</Link>
								</li>
								<li role="separator" className="divider" />
								<li>
									<Link to="/logout">Logout</Link>
								</li>
							</ul>
						</li>
					</ul>
				</div>
			</div>
		</nav>
	);
};

export default NavBar;
