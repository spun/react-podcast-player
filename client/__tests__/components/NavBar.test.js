import React from "react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
// Component under test
import NavBar from "../../src/js/components/NavBar";

test("NavBar creation", () => {
	const userEmail = "demoEmail";

	const component = renderer.create(
		<MemoryRouter>
			<NavBar userName={userEmail} />
		</MemoryRouter>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
