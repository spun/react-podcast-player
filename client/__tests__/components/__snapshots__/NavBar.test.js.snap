// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar creation 1`] = `
<nav
  className="navbar navbar-default"
>
  <div
    className="container-fluid"
  >
    <div
      className="navbar-header"
    >
      <button
        className="navbar-toggle"
        type="button"
      >
        <span
          className="sr-only"
        >
          Toggle navigation
        </span>
        <span
          className="icon-bar bar1"
        />
        <span
          className="icon-bar bar2"
        />
        <span
          className="icon-bar bar3"
        />
      </button>
      <a
        className="navbar-brand"
        href="#"
      >
        Dashboard
      </a>
    </div>
    <div
      className="collapse navbar-collapse"
    >
      <ul
        className="nav navbar-nav navbar-right"
      >
        <li>
          <a
            href="/add"
            onClick={[Function]}
          >
            <i
              className="ti-plus"
            />
             
            <p>
              Add suscription
            </p>
          </a>
        </li>
        <li
          className="dropdown"
        >
          <a
            className="dropdown-toggle"
            data-toggle="dropdown"
            href="#"
          >
            <i
              className="ti-settings"
            />
             
            <p>
              Settings
            </p>
            <b
              className="caret"
            />
          </a>
          <ul
            className="dropdown-menu"
          >
            <li>
              <a
                href="/account"
                onClick={[Function]}
              >
                My account
              </a>
            </li>
            <li>
              <a
                href="/about"
                onClick={[Function]}
              >
                About
              </a>
            </li>
            <li
              className="divider"
              role="separator"
            />
            <li>
              <a
                href="/logout"
                onClick={[Function]}
              >
                Logout
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
